src/global/tracking/TrackParamTruthInit_factory.cc

```diff
@@ -15,12 +15,20 @@
 #include <algorithms/tracking/TrackParamTruthInitConfig.h>
 
 void eicrecon::TrackParamTruthInit_factory::Init() {
+    auto app = GetApplication();
+
     // This prefix will be used for parameters
     std::string param_prefix = GetDefaultParameterPrefix();
 
     InitLogger(param_prefix);
     InitDataTags(param_prefix);
 
+    // Algorithm configuration
+    m_truth_track_seeding_algo.m_momentum_split = 0.0;
```

Comment:
It is suggested, instead of using class memebers directly (which is supposed to be somewhat prived) it is suggested to use <className>Config POD structure to store algorithms configs. And then you can inherit from WithPODConfig to get some helper methods. To be more precise: 

- [Example of simple config TrackerHitReconstructionConfig.h](https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/TrackerHitReconstructionConfig.h)
- [Example of CFKTrackingConfig](https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/CKFTrackingConfig.h)
   
   ```c++
       // Initialized with default values
       struct CKFTrackingConfig {
        std::vector<double> m_etaBins = {};  // {this, "etaBins", {}};
        std::vector<double> m_chi2CutOff = {50.}; //{this, "chi2CutOff", {15.}};
        std::vector<size_t> m_numMeasurementsCutOff = {10}; //{this, "numMeasurementsCutOff", {10}};
    };
   ```
   
- [Example of using WithPODConfig for CFKTracking](https://github.com/eic/EICrecon/blob/cb09f2e59edaab38e3ce348a64d9ef01c4328531/src/algorithms/tracking/CKFTracking.h#L45)

   ```c++
       class CKFTracking: public WithPodConfig<eicrecon::CKFTrackingConfig> {
   ```
   
   It makes m_cfg (or getConfig()) available inside CKFTracking.

- Then if EICRecon specific JChainFactoryT is used (and they are used everywhere for tracking in EICrecon) it also provides helper methods. [Example with JChainFactoryT  definition for CKFTracking:](https://github.com/eic/EICrecon/blob/cb09f2e59edaab38e3ce348a64d9ef01c4328531/src/global/tracking/CKFTracking_factory.h#L21) 

   ```c++
       class CKFTracking_factory :
            public JChainFactoryT<eicrecon::TrackingResultTrajectory, CKFTrackingConfig>,
            public SpdlogMixin<CKFTracking_factory>
   ```

- Config parameters are simply accessible [example](https://github.com/eic/EICrecon/blob/main/src/global/tracking/TrackerHitReconstruction_factory.cc#L19)

   ```c++
       // Set time resolution
    app->SetDefaultParameter(param_prefix + ":TimeResolution", m_reco_algo.getConfig().time_resolution, "threshold");
    ```
  
- For more complex cases DefaultConfig() and then ApplyConfig methods could be used. [Example](https://github.com/eic/EICrecon/blob/cb09f2e59edaab38e3ce348a64d9ef01c4328531/src/global/tracking/CKFTracking_factory.cc#L40)

   ```c++
       // Algorithm configuration
    auto cfg = GetDefaultConfig();
    app->SetDefaultParameter(param_prefix + ":EtaBins", cfg.m_etaBins, "Eta Bins for ACTS CKF tracking reco");
    app->SetDefaultParameter(param_prefix + ":Chi2CutOff", cfg.m_chi2CutOff, "Chi2 Cut Off for ACTS CKF tracking");
    app->SetDefaultParameter(param_prefix + ":NumMeasurementsCutOff", cfg.m_numMeasurementsCutOff, "Number of measurements Cut Off for ACTS CKF tracking");

    // Initialize algorithm
    m_tracking_algo.applyConfig(cfg);
  ```

Reply:
Out of scope of this PR.